jitify2.hpp

```diff
@@ -5593,397 +5666,1599 @@ inline bool read_text_file(const std::string& fullpath, std::string* content) {
+        return token(match('>') ? match('=') ? Tt::kGtGtEq : Tt::kGtGt
+                                : match('=') ? Tt::kGtEq : Tt::kGt);
+      case ':': return token(match('>')
+                           ? Tt::kRBracket
+                           : match(':') ? Tt::kColonColon : Tt::kColon);
+      case '{': return token(Tt::kLBrace);
+      case '}': return token(Tt::kRBrace);
+      case '%':
+        return token(match('>')
+                         ? Tt::kRBrace
+                         : match(':')
+                               // TODO: Probably need to do the in_directive_
+                               // etc. logic here too.
+                               ? match("%:") ? Tt::kHashHash : Tt::kHash
+                               : match('=') ? Tt::kPercentEq : Tt::kPercent);
+      case '.': return token(match('*') ? Tt::kDotStar : Tt::kDot);
+      case '-': return token(match('>') ? match('*') ? Tt::kArrowStar
+                                                     : Tt::kArrow
+                                        : match('-') ? Tt::kMinusMinus
+                                                     : match('=') ? Tt::kMinusEq
+                                                                  : Tt::kMinus);
+
+      case ',': return token(Tt::kComma);
+      case '+': return token(match('+') ? Tt::kPlusPlus
+                                        : match('=') ? Tt::kPlusEq
+                                                     : Tt::kPlus);
+      case '*': return token(match('=') ? Tt::kStarEq : Tt::kStar);
+      case '/': return match('/')
+                           ? line_comment()
+                           : match('*') ? block_comment()
+                                        : token(match('=') ? Tt::kSlashEq
+                                                           : Tt::kSlash);
+      // Note: Trigraphs not supported.
+      case '?': return token(Tt::kQuestion);
+      case '&': return token(match('&') ? Tt::kAmpAmp
+                                        : match('=') ? Tt::kAmpEq : Tt::kAmp);
+      case '|': return token(match('|') ? Tt::kBarBar
+                                        : match('=') ? Tt::kBarEq : Tt::kBar);
+      case '^': return token(match('=') ? Tt::kCaretEq : Tt::kCaret);
+      case '~': return token(Tt::kTilde);
+      case '=': return token(match('=') ? Tt::kEqEq : Tt::kEq);
+      case '!': return token(match('=') ? Tt::kBangEq : Tt::kBang);
+      case '#':
+        return token(match('#') ? Tt::kHashHash
+                                : (is_start_of_directive_ = !in_directive_,
+                                   in_directive_ = true,
+                                   Tt::kHash));
+      case '\'': return character();
+      case '"': return in_include_directive_ ? quote_include() : string();
+      case 'u': match('8');
+        // fall-through
+        [[gnu::fallthrough]];  // Not sure why gcc complains here without this
```

Comment:
```suggestion
#ifdef __GNUC__
        [[gnu::fallthrough]];  // Not sure why gcc complains here without this
#endif
```
*Visual Studio 2022*
`jitify2.hpp(6192): warning C5030: attribute 'gnu::fallthrough' is not recognized`